crates/lib/src/keyboard.rs

```diff
@@ -0,0 +1,78 @@
+use async_priority_channel as priority;
+use tokio::{io::AsyncBufReadExt, sync::mpsc};
+use tracing::trace;
+
+use crate::{
+	error::{CriticalError, RuntimeError},
+	event::{Event, Priority, Source, Tag},
+};
+
+#[derive(Debug, Clone, Default)]
+pub struct WorkingData {
+	pub eof: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Keyboard {
+	Eof,
+}
+
+pub async fn worker(
+	errors: mpsc::Sender<RuntimeError>,
+	events: priority::Sender<Event, Priority>,
+) -> Result<(), CriticalError> {
+	imp_worker(errors, events).await
```

Comment:
config handling: this function essentially should do a loop that watches the working data. if the bool changes to true, start up an `imp_worker` task and keep a handle, if it changes to false, cancel the task (or send a cancel, we want to drop the stdin held within the task and not leave it dangling), otherwise do nothing and wait more.

Reply:
I haven't tackled this yet. The code now included something that watches for reconfigurations but it won't interrupt the stdin monitoring to do it. I'll try to look at this soon.

Reply:
I've made an attempt at this. I think it is reasonable but wouldn't be surprised if there was a nicer way to achieve it.